net/net.go

```diff
@@ -1178,6 +1223,10 @@ func (n *net) startPulling() {
 	// it will be redefined on the next iteration
 	var interval = InitialPullInterval
 
+	// group threads by peers and exchange edges efficiently
+	var compressor = queue.NewThreadPacker(n.ctx, MaxThreadsExchanged, ExchangeCompressionTimeout)
+	go n.startExchange(compressor)
+
 PullCycle:
 	for {
 		ts, err := n.store.Threads()
```

Comment:
It would be great to distribute this over some time interval by paginating over the threads in an efficient manner. When there are 10s of 1000s of threads, this is a pretty heavy call.